postgres-appliance/scripts/configure_spilo.py

```diff
@@ -88,8 +88,23 @@ def write_certificates(environment, overwrite):
 
     ssl_keys = ['SSL_CERTIFICATE', 'SSL_PRIVATE_KEY']
     if set(ssl_keys) <= set(environment):
+        logging.info('Writing custom ssl certificate')
         for k in ssl_keys:
             write_file(environment[k], environment[k + '_FILE'], overwrite)
+        if 'SSL_CA' in environment:
+            if environment['SSL_CA_FILE'] == '':
+                environment['SSL_CA_FILE'] = os.path.join(environment['RW_DIR'], 'certs', 'ca.crt')
```

Comment:
Unfortunately, the `write_certificates()` function is called after the `TEMPLATE` is evaluated and hence if `SSL_CA` is set, but `SSL_CA_FILE` isn't set, the file will be written, but postgres.yml will lack `postgresql.parameters.ssl_ca_file`.

The same applies to `SSL_CRL_FILE`.

Such checks should be moved to the `get_placeholders()` function.

Reply:
ok